### PR TITLE
Add coverage for logical ors and ands in const_assert

### LINKED ISSUE #2750: wgsl: expand coverage in const_assert.spec.ts
A [bug](https://bugs.chromium.org/p/tint/issues/detail?id=1961) in Tint was discovered while working on AbstractFloat tests, which indicates that CTS coverage of const_assert is insufficient.

Specifically making sure that things like short circuiting and ordering are covered.

### PR DESCRIPTION
**Pull request by zoddicus**
This also refactors a bunch of the testing code to not have both tests for asserting and not asserting in the same stanza, since it makes the logs clearer about what expectation is failing

Fixes #2750

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [ ] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

**Comment by zoddicus** [comment]
PTAL, some of these tests current fail on Tint due to a [KI](https://bugs.chromium.org/p/tint/issues/detail?id=1961), but they fail in the way that I expect.

File: src/webgpu/shader/validation/const_assert/const_assert.spec.ts [hunk 1]
```diff
@@ -7,44 +7,174 @@ import { ShaderValidationTest } from '../shader_validation_test.js';
...
+    const val = kConditionCases[t.params.lhs].val || kConditionCases[t.params.rhs].val;
+    t.expectCompileResult(
+      false,
+      kConditionConstants + buildStaticAssert(val ? `!(${expr})` : expr, t.params.scope)
+    );
+  });
+
+g.test('constant_expression_logical_and_no_assert')
+  .desc(
+    `Test that const_assert does not assert on a condition expression that contains a logical and and evaluates true`
```
Comment (ben-clayton) [inline src/webgpu/shader/validation/const_assert/const_assert.spec.ts:138]: nit: this sentence is hard to parse. Recommend:

`Test that const_assert does not assert on a condition expression that contains a logical-and which evaluates to true`

Similarly for the other comments.

**Review by ben-clayton — APPROVED** [review]
One minor nit, otherwise looks great!

File: src/webgpu/shader/validation/const_assert/const_assert.spec.ts (hunk 1 above)
Comment (zoddicus) [reply src/webgpu/shader/validation/const_assert/const_assert.spec.ts]: Done
